docs/tech-reference/rsync-distributor.rst

```diff
@@ -0,0 +1,105 @@
+=====================
+RPM rsync Distributor
+=====================
+
+Purpose:
+========
+The RPM rsync distributor publishes RPM content to a remote server. As the name suggests, the
+distributor uses rsync to copy.
+
+Configuration
+=============
+Here's example of rpm_rsync_distributor configuration:
+
+.. code-block:: json
+
+    {
+     "distributor_id": "rpm_rsync_distributor",
+     "distributor_type_id": "rpm_rsync_distributor",
+     "distributor_config": {
+        "remote": {
+            "auth_type": "publickey",
+            "ssh_login": "foo",
+            "key_path": "/home/user/.ssh/id_rsa",
+            "host": "192.168.121.1",
+            "root": "/home/foo/pulp_root_dir"
+        },
+        "predistributor_id": "yum_distributor",
+        "force_full": true
+        }
+    }
+
+``auth_type``
+  Two authentication methods are supported: ``publickey`` and ``password``.
+
+``ssh_login``
+  ssh login to remote machine
+
+``key_path``
+  A path to the private key that will be used as identity file for ssh. When ``auth_type`` is
+  'publickey' this is a required config. The key has to be owned by user and group `apache`.
+
+``ssh_password``
+  Password is required when ``auth_type`` is 'password'.
+
+``host``
+  remote hostname
+
+``root``
+  Remote root directory with all the data (content and published contnet). Remote equivalent to
+  ``/var/lib/pulp``.
+
+Optional configuration
+----------------------
+
+``skip_fast_forward``
+  If true, rsync distributor will publish all content of repository.
+
+``origin_only``
+  If true, the distributor will publish hard content only (e.g. /var/lib/pulp/content)
+
+``skip_repodata``
+  If true, repodata will be omitted from the publish.
+
+``force_refresh``
+  If true, appends --delete to the rsync command for symlinks and repodata so any old files no longer present in
+  the local published directory are removed from remote server.
+
+
+How RPM rsync distributor works
```

Comment:
Is it a good idea to list the internal implementation of the RPM rsync distributor in its documentation? Pulp doesn't really have requirements documents, which means that this documentation serves as the contract between users and the application. I don't think users should be aware of facts such as that Pulp workers create symlinks in `/var/cache/pulp/worker1/taskuuid/.relative`. Listing this information here also makes it harder for readers to find the information they care about, such as whether the rsync distributor can copy to local directories or not, which transports may be used, what data is copied to the destination directory, and the relationship is between the RPM rsync distributor and the other RPM distributors. (What is a predistributor or postdistributor? If those details are handled by Pulp, then do I need to know about them?) And more importantly, listing this information here makes it harder for us to change Pulp's internal implementation of these features, as doing so means breaking a contract.

Can we gear this documentation toward Pulp users?